lms/services/application_instance.py

```diff
@@ -0,0 +1,54 @@
+from typing import Optional
+
+from sqlalchemy import or_
+
+from lms.models import ApplicationInstance
+
+
+class ApplicationInstanceService:
+    def __init__(self, db):
+        self._db = db
+
+    def find(self, query: str) -> Optional[ApplicationInstance]:
```

Comment:
I not sure if this belong here, should this type of query go into the model itself? Then will need boilerplate code here on the service to call them 1 service method per 1 model one.